specs/Proposals/0040-split-states.md

```diff
@@ -206,52 +271,46 @@ fn apply_chunks(...) -> Result<(), Error> {
+  - It is the best because its implementation is the simplest.
+    Considering we want to launch Simple Nightshade as soon as possible by Q4 2021 and we will not enable challenges any time soon, this is the best option we have.
 - What other designs have been considered and what is the rationale for not choosing them?
+  - We have considered other designs that changes states incrementally and keep state roots on chain to make it compatible with challenges.
+However, the implementaion of those approaches are overly complicated and does not fit into our timeline for launching Simple Nightshade.
 - What is the impact of not doing this?
+  - The impact will be the delay of launching Simple Nightshade, or no launch at all.
 
 # Unresolved questions
 [unresolved-questions]: #unresolved-questions
-#TODO
-
 - What parts of the design do you expect to resolve through the NEP process before this gets merged?
+  - None. The NEP is quite thorough already.
 - What parts of the design do you expect to resolve through the implementation of this feature before stabilization?
+  - There might be small changes in the detailed implemenations or specifications of some of the functions described above, but the overall structure will not be changed.
 - What related issues do you consider out of scope for this NEP that could be addressed in the future independently of the solution that comes out of this NEP?
+  - One issue that is related to this NEP but will be resolved indepdently is how trie nodes are stored in the database.
+    Right now, it is a combination of `shard_id` and the node hash.
+    Part of the change proposed in this NEP regarding `ShardId` is because of this.
+    Plans on how to only store the node hash as keys are being discussed [here](https://github.com/near/nearcore/issues/4527), but it will happen after the Simple Nightshade migration since completely solving the issue will take some careful design and we want to prioritize launching Simple Nightshade for now.
 
 # Future possibilities
 [future-possibilities]: #future-possibilities
-
-Think about what the natural extension and evolution of your proposal would
-be and how it would affect the project as a whole in a holistic
-way. Try to use this section as a tool to more fully consider all possible
-interactions with the project in your proposal.
-Also consider how the this all fits into the roadmap for the project
-and of the relevant sub-team.
-
-This is also a good place to "dump ideas", if they are out of scope for the
-NEP you are writing but otherwise related.
-
-If you have tried and cannot think of any future possibilities,
-you may simply state that you cannot think of anything.
-
-Note that having something written down in the future-possibilities section
-is not a reason to accept the current or a future NEP. Such notes should be
-in the section on motivation or rationale in this or subsequent NEPs.
-The section merely provides additional information.
+None
```

Comment:
well we've discussed possibly switching to on-chain state updates once we introduce challenges...